doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -454,6 +454,37 @@ Field              | Type       | Meaning
 `cpu_nano_cores`   | [`bigint`] | Approximate CPU usage, in billionths of a vCPU core.
 `memory_bytes`     | [`bigint`] | Approximate RAM usage, in bytes.
 
+### `mz_storage_source_statistics`
+
+The `mz_storage_source_statistics` table gives the last known storage-specific statistics
+for all processes of all extant storage replicas.
+
+At this time, we do not make any guarantees about the exactness or freshness of these numbers.
+
+Field                 | Type        | Meaning
+----------------------|-------------|--------
+`id`                 | [`text`]     | The ID of the storage object (source or sink).
+`worker_id`          | [`bigint`]   | An identifier of a process within a host.
+`snapshot_committed` | [`boolean`]  | Whether or not we have committed the initial snapshot for a source.
+`messages_received`  | [`bigint`]   | How many messages we have received from upstream.
+`messages_committed` | [`bigint`]   | How many messages we have committed into our storage layer.
+`bytes_received`     | [`bigint`]   | How many bytes worth of messages we have received from upstream.
+
+These values all require some amount of interpretation:
```

Comment:
Very thoughtful! \o/

let's talk in person but I think this is close to what we want. Or it!